diff --git a/drivers/i2c/i2c-dev.c b/drivers/i2c/i2c-dev.c
index 38d9d3de2..1fd6cf848 100644
--- a/drivers/i2c/i2c-dev.c
+++ b/drivers/i2c/i2c-dev.c
@@ -35,6 +35,7 @@
 #define I2C2 "i2c@04020000"
 #define I2C3 "i2c@04030000"
 #define I2C4 "i2c@04040000"
+#define I2C5 "i2c5@gpio"
 #endif

 /*
@@ -680,6 +681,8 @@ static int i2cdev_attach_adapter(struct device *dev, void *dummy)
                adap->i2c_idx = 3;
        } else if (!strcmp(adap->dev.of_node->full_name, I2C4)) {
                adap->i2c_idx = 4;
+       } else if (!strcmp(adap->dev.of_node->full_name, I2C5)) {
+               adap->i2c_idx = 5;
        } else {
                adap->i2c_idx = adap->nr;
        }
